### PR TITLE
fix(replays): Fix race condition when loading replay errors & attachments

### LINKED ISSUE #38852: [jest] unit test useReplayData
- [ ] checking error states, urls requested
- [ ] look for edge cases

### LINKED ISSUE #45224: Error isn't showing in replay
### Environment

SaaS (https://sentry.io/)

### Version

_No response_

### Link

https://sentry.sentry.io/replays/javascript:2dfcb09f8efd455ab0627514aa2ece5d/?query=&referrer=%2Freplays%2F&statsPeriod=14d&t_main=console&yAxis=count%28%29

### DSN

_No response_

### Steps to Reproduce

Go to a replay that says it has an error.

### Expected Result

See the error in console / breadcrumbs.

### Actual Result

error isn't in console / breadcrumbs,but can see in issues tab.

### PR DESCRIPTION
**Pull request by ryan953**
This PR fixes an issue where error data doesn't get set into state if that payload is ready before the first attachments payload.

I've highlighted the actual problem in the original code in the PR, but I also took the chance to stream line everything so state is more separate, and we rely only on react useCallback and useEffect hooks to trigger fetch requests.

Tested incremental loading against the huge replay `/replays/javascript:23003c384e0c4fc084c44ea906240a4a/` 
Also tested against `/replays/javascript:2dfcb09f8efd455ab0627514aa2ece5d/` which loads one error
And finally a replay that has no errors to load: `/replays/javascript:f927296182c74dd3bcf4d089a3c819ac/`
All in the sentry org.

Fixes #45224

### CODE REVIEW

**Review by JoshFerge — COMMENTED** [review]
is there any way to test this regression in like an acceptance test or something?

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 1]
```diff
@@ -96,143 +81,119 @@ function useReplayData({replaySlug, orgSlug}: Options): Result {
...
-                end: finishedAtClone.toISOString(),
-                query: `id:[${String(errorIds)}]`,
-              },
-            }
-          );
-          promise.then(response => {
-            setState(prev => {
-              if (!prev.attachments?.length) {
-                return prev;
-              }
```
Comment (ryan953) [inline static/app/utils/replays/hooks/useReplayData.tsx:173]: This was a dumb check i think. It meant that we had to start with attachment data, which shouldn't matter when we're fetching error info. 

It looks like it was a hold-over from earlier when fetching was more sequential.

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 2]
```diff
@@ -47,25 +37,25 @@ type Options = {
...
+  fetchingReplay: true,
 });
 
-function responsesToAttachments(responses: Array<unknown>) {
-  // Each response returns an array of segments
+function responseToAttachments(segment: unknown[]) {
   // Each segment includes an array of attachments
   // Therefore we flatten 2 levels deep
-  return responses.flat(2);
+  return segment.flat(1);
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayData.tsx:58]: ack, should be 1 now. We're not passing in the extra array wrapper anymore.

**Comment by ryan953** [comment]
i'm going to merge this in because it's an issue in prod. Then follow up with more unit tests. I think i figured out how to artificially defer api requests under test, so it should be possible to test incremental loading states

**Pull request merged**